contracts/transfer/circuits/src/gadgets/merkle.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) DUSK NETWORK. All rights reserved.
+// Licensed under the MPL 2.0 license. See LICENSE file in the project root for details. 
+
+use phoenix_core::note::Note;
+use dusk_plonk::prelude::*;
+use poseidon252::merkle_proof::{merkle_opening_gadget, PoseidonBranch};
+use plonk_gadgets::AllocatedScalar;
+
+
+pub fn merkle(composer: &mut StandardComposer, branch: PoseidonBranch, note_hash: AllocatedScalar) {
+
+    let leaf = note_hash.var;
+    let root = branch.root;
+
+    merkle_opening_gadget(composer, branch, leaf, root);
+}
```

Comment:
As I said this does not make sense to me. Having a function wraping just another function.

@Bounce23 told me that @ZER0 and @toghrulmaharramov said that it should be like this. i don't agree. So we should discuss it. But for now if they said that, it's fine.